vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleMainComponent.java

```diff
@@ -0,0 +1,140 @@
+import life.qbic.projectmanagement.domain.project.ProjectId;
+import life.qbic.projectmanagement.domain.project.experiment.Experiment;
+import life.qbic.projectmanagement.domain.project.sample.Batch;
+import life.qbic.projectmanagement.domain.project.sample.Sample;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Sample main component
+ * <p>
+ * The main component is a {@link Div} container, which is responsible for hosting the components
+ * handling the main content within the {@link SampleInformationPage}. It propagates the
+ * {@link Project} and {@link Experiment} to the components within this container. Additionally, it
+ * propagates the {@link Batch} and {@link Sample} information provided in the
+ * {@link SampleOverviewComponent} to the {@link SampleInformationPage} and can be easily extended
+ * with additional components.
+ */
+@SpringComponent
+@UIScope
+@PermitAll
+public class SampleMainComponent extends Div {
+
+  @Serial
+  private static final long serialVersionUID = -5431288053780884294L;
+  private ProjectId projectId;
+  private static final Logger log = LoggerFactory.logger(SampleMainComponent.class);
+  private final SampleOverviewComponent sampleOverviewComponent;
+  private final transient ProjectInformationService projectInformationService;
+  private final transient ExperimentInformationService experimentInformationService;
+
+  public SampleMainComponent(@Autowired ProjectInformationService projectInformationService,
+      @Autowired ExperimentInformationService experimentInformationService,
+      @Autowired SampleOverviewComponent sampleOverviewComponent) {
+    this.addClassName("main");
+    this.sampleOverviewComponent = sampleOverviewComponent;
+    this.projectInformationService = projectInformationService;
+    this.experimentInformationService = experimentInformationService;
+    reloadOnBatchRegistration();
+  }
+
+  /**
+   * Provides the {@link ProjectId} to the components within this container
+   * <p>
+   * This method serves as an entry point providing the necessary {@link ProjectId} to components
+   * within this container
+   *
+   * @param projectId projectId of the selected project
+   */
+  public void projectId(ProjectId projectId) {
+    this.projectId = projectId;
+    projectInformationService.find(projectId)
+        .ifPresentOrElse(project -> {
+          propagateProjectInformation(projectId);
+          propagateExperimentInformation(projectId);
+        }, this::displayProjectNotFound);
+  }
+
+  private void propagateProjectInformation(ProjectId projectId) {
+    sampleOverviewComponent.setProject(projectId);
+  }
+
+  private void propagateExperimentInformation(ProjectId projectId) {
+    Collection<Experiment> experiments = getExperimentsForProject(projectId);
+    if (experiments.isEmpty()) {
+      displayNoExperimentsFound();
+    } else {
+      displayComponentInContent(sampleOverviewComponent);
```

Comment:
is this really as abstract as the name suggests? otherwise I would suggest "displaySampleOverview(...)" or similar